File: backend/crawl/app/models/team_pitcher_stats.py
from sqlalchemy import Column, Integer, String, ForeignKey, Float
from sqlalchemy.orm import relationship
from app.models.database import Base


class TeamPitcherStats(Base):
    __tablename__ = "team_pitcher_stats"

    id = Column(Integer, primary_key=True, index=True)
    rank = Column(Integer)
    year = Column(String(10))
    team_id = Column(Integer, ForeignKey("team.id"))
    era = Column(String(10))
    game_count = Column(Integer)
    win = Column(Integer)
    lose = Column(Integer)
    save = Column(Integer)
    hld = Column(Integer)
    wpct = Column(String(10))
    ip = Column(String(10))
    h = Column(Integer)
    hr = Column(Integer)
    bb = Column(Integer)
    hbp = Column(Integer)
    so = Column(Integer)
    r = Column(Integer)
    er = Column(Integer)
    whip = Column(String(10))
    
    team = relationship("Team", back_populates="team_pitcher_stats")

File: backend/baseball/src/main/java/com/newbie/baseball/domain/teamStats/entity/TeamPitcherStats.java
package com.newbie.baseball.domain.teamStats.entity;

import com.newbie.baseball.domain.team.entity.Team;
import jakarta.persistence.*;
import lombok.Getter;

@Getter
@Entity
@Table(name = "team_pitcher_stats")
public class TeamPitcherStats {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Integer id;

    @Column(name = "rank")
    private Integer rank;

    @Column(name = "year", length = 10)
    private String year;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "team_id")
    private Team team;

    @Column(name = "era", length = 10)
    private String era;

    @Column(name = "game_count")
    private Integer gameCount;

    @Column(name = "win")
    private Integer win;

    @Column(name = "lose")
    private Integer lose;

    @Column(name = "save")
    private Integer save;

    @Column(name = "hld")
    private Integer hld;

    @Column(name = "wpct", length = 10)
    private String wpct;

    @Column(name = "ip", length = 10)
    private String ip;

    @Column(name = "h")
    private Integer h;

    @Column(name = "hr")
    private Integer hr;

    @Column(name = "bb")
    private Integer bb;

    @Column(name = "hbp")
    private Integer hbp;

    @Column(name = "so")
    private Integer so;

    @Column(name = "r")
    private Integer r;

    @Column(name = "er")
    private Integer er;

    @Column(name = "whip", length = 10)
    private String whip;

}
